pkg/autodiscovery/listeners/snmp_test.go

```diff
@@ -57,6 +59,62 @@ func TestSNMPListener(t *testing.T) {
 	assert.Equal(t, "192.168.0.0", job.subnet.startingIP.String())
 }
 
+func TestSNMPListenerSubnets(t *testing.T) {
+	newSvc := make(chan Service, 10)
+	delSvc := make(chan Service, 10)
+	testChan := make(chan snmpJob)
+
+	listenerConfig := snmp.ListenerConfig{
+		Configs: []snmp.Config{},
+		Workers: 10,
+	}
+
+	for i := 0; i < 100; i++ {
+		snmpConfig := snmp.Config{
+			Network:     "172.18.0.0/30",
+			Community:   "f5-big-ip",
+			Port:        1161,
+			ContextName: "context" + strconv.Itoa(i),
+		}
+		listenerConfig.Configs = append(listenerConfig.Configs, snmpConfig)
+	}
+
+	mockConfig := config.Mock()
+	mockConfig.Set("snmp_listener", listenerConfig)
+
+	worker = func(l *SNMPListener, jobs <-chan snmpJob) {
+		for {
+			job := <-jobs
+			testChan <- job
+		}
+	}
+
+	snmpListenerConfig, err := snmp.NewListenerConfig()
+	assert.Equal(t, nil, err)
+
+	services := map[string]Service{}
+	l := &SNMPListener{
+		services: services,
+		stop:     make(chan bool),
+		config:   snmpListenerConfig,
+	}
+
+	l.Listen(newSvc, delSvc)
+
+	subnets := make(map[uintptr]bool)
+	entities := make(map[string]bool)
+
+	for i := 0; i < 400; i++ {
+		job := <-testChan
+		subnets[uintptr(unsafe.Pointer(job.subnet))] = true
+		entities[job.subnet.config.Digest(job.currentIP.String())] = true
+	}
+
+	// make sure we have 100 subnets and 400 different entity hashes
+	assert.Equal(t, 100, len(subnets))
+	assert.Equal(t, 400, len(entities))
```

Comment:
Note for reviewers: Previously, the number of entities varied from 200 to 300 at each run. The `job` retrieved from `testChan` was not pointing to the right `job.subnet`, the variation is probably due to concurrency.